### PR TITLE
Better logger support

### PR DESCRIPTION
**Pull request by MattBDev**
This PR removed the `debug()` methods in IFawe.java and Fawe.java and replaces their usages with a proper logger.
It also removes (almost) all usages of System.out or System.err and replaces them with a proper logger.
Some new messages were added for debugging purposes.

### CODE REVIEW

**Comment by N0tMyFaultOG** [comment]
If we are already using slf4j, why don't we do something like 
`private static final Logger logger = LoggerFactory.getLogger("FAWE/" + ResidenceFeature.class.getSimpleName());` for the appropriate class
and call sth like logger.info("Plugin 'Residence' found. Using it now."). Would look a bit cleaner imo.

**Comment by MattBDev** [comment]
> If we are already using slf4j, why don't we do something like
> `private static final Logger logger = LoggerFactory.getLogger("FAWE/" + ResidenceFeature.class.getSimpleName());` for the appropriate class
> and call sth like logger.info("Plugin 'Residence' found. Using it now."). Would look a bit cleaner imo.

@IntellectualSites/fawe-team thoughts?

**Comment by N0tMyFaultOG** [comment]
> > If we are already using slf4j, why don't we do something like
> > `private static final Logger logger = LoggerFactory.getLogger("FAWE/" + ResidenceFeature.class.getSimpleName());` for the appropriate class
> > and call sth like logger.info("Plugin 'Residence' found. Using it now."). Would look a bit cleaner imo.
> 
> @IntellectualSites/fawe-team thoughts?

Actually, that was just a quick thought I had during making dinner lol.
But checking upstream, they do it the same way I thought about it. This could potentially help preventing merge conflicts and upstream compatibility in the future.

**Review by dordsor21 — APPROVED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/regions/EllipsoidRegion.java [hunk 1]
```diff
@@ -409,7 +409,7 @@ public void filter(IChunk chunk, Filter filter, ChunkFilterBlock block, IChunkGe
             int yTopFull = Math.min(255, cy + diffYFull);
 
             if (yBotFull == yTopFull || yBotFull > yTopFull) {
-                System.out.println("aa");
+                System.out.println("aa"); //why do we print "aa" here?
```
Comment (N0tMyFaultOG) [inline worldedit-core/src/main/java/com/sk89q/worldedit/regions/EllipsoidRegion.java:412]: Looks like some left over debug?

**Review by N0tMyFaultOG — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/regions/EllipsoidRegion.java (hunk 1 above)
Comment (MattBDev) [reply worldedit-core/src/main/java/com/sk89q/worldedit/regions/EllipsoidRegion.java:412]: Yeah, that's what I thought but what could it mean and is it safe to delete?

Comment (N0tMyFaultOG) [reply worldedit-core/src/main/java/com/sk89q/worldedit/regions/EllipsoidRegion.java:412]: I'd say so.

Comment (N0tMyFaultOG) [reply worldedit-core/src/main/java/com/sk89q/worldedit/regions/EllipsoidRegion.java:412]:
```suggestion
```

**Review by N0tMyFaultOG — APPROVED** [review]

**Pull request merged**